### ISSUE TITLE
[Bug]: Problems with the latest version

### ISSUE THREAD

**Issue by Qee1y** [someaddons/dungeonbosses#72]
### Describe the bug you're experiencing

[2024-03-04-7.log](https://github.com/someaddons/dungeonbosses/files/14477407/2024-03-04-7.log)

### Reproducability

Line 7536
 The world is stuck at 3%

### Mod up to date

- [X] Before submitting this issue I updated to the newest version and reproduced it

### Minecraft version

1.18

### Modloader version

Forge

### Logs and Datapacks

no packet

**Comment by Qee1y** [comment]
mc1.20.1 

I don't know why it hasn't been changed to the version

**Comment by SiverDX** [comment]
Same issue here for 1.19.2 (not on world start but server timed out due to the boss trying to find a spawn position)

```
"Worker-Main-203" daemon prio=4 Id=608 WAITING on java.util.concurrent.CompletableFuture$Signaller@311a8456
	at java.base@17.0.8/jdk.internal.misc.Unsafe.park(Native Method)
	-  waiting on java.util.concurrent.CompletableFuture$Signaller@311a8456
	at java.base@17.0.8/java.util.concurrent.locks.LockSupport.park(LockSupport.java:211)
	at java.base@17.0.8/java.util.concurrent.CompletableFuture$Signaller.block(CompletableFuture.java:1864)
	at java.base@17.0.8/java.util.concurrent.ForkJoinPool.compensatedBlock(ForkJoinPool.java:3451)
	at java.base@17.0.8/java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3434)
	at java.base@17.0.8/java.util.concurrent.CompletableFuture.waitingGet(CompletableFuture.java:1898)
	at java.base@17.0.8/java.util.concurrent.CompletableFuture.join(CompletableFuture.java:2117)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ServerChunkCache.m_7587_(ServerChunkCache.java:113)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.Level.m_6522_(Level.java:179)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.LevelReader.m_46819_(LevelReader.java:129)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.Level.m_6325_(Level.java:174)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.Level.m_8055_(Level.java:352)
	at TRANSFORMER/brutalbosses@1.19.2-6.7/com.brutalbosses.entity.BossSpawnHandler.lambda$findSpawnPosForBoss$0(BossSpawnHandler.java:146)
	at TRANSFORMER/brutalbosses@1.19.2-6.7/com.brutalbosses.entity.BossSpawnHandler$$Lambda$38107/0x00007f7e74f7a428.test(Unknown Source)
	at TRANSFORMER/cupboard@1.19.2-2.3/com.cupboard.util.BlockSearch.findAround(BlockSearch.java:57)
	at TRANSFORMER/brutalbosses@1.19.2-6.7/com.brutalbosses.entity.BossSpawnHandler.findSpawnPosForBoss(BossSpawnHandler.java:143)
	at TRANSFORMER/brutalbosses@1.19.2-6.7/com.brutalbosses.entity.BossSpawnHandler.spawnBoss(BossSpawnHandler.java:111)
	at TRANSFORMER/brutalbosses@1.19.2-6.7/com.brutalbosses.entity.BossSpawnHandler.onChestPlaced(BossSpawnHandler.java:55)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.block.entity.RandomizableContainerBlockEntity.handler$emj000$onSetLoot(RandomizableContainerBlockEntity.java:534)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.block.entity.RandomizableContainerBlockEntity.m_59626_(RandomizableContainerBlockEntity.java:89)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.block.entity.RandomizableContainerBlockEntity.m_222766_(RandomizableContainerBlockEntity.java:42)
	at TRANSFORMER/apotheosis@6.4.1/shadows.apotheosis.adventure.gen.BossDungeonFeature.m_142674_(BossDungeonFeature.java:114)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.levelgen.feature.Feature.m_225028_(Feature.java:154)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.levelgen.feature.ConfiguredFeature.m_224953_(ConfiguredFeature.java:27)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.levelgen.placement.PlacedFeature.m_226362_(PlacedFeature.java:56)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.levelgen.placement.PlacedFeature$$Lambda$37038/0x00007f7e77f81d80.accept(Unknown Source)
	at java.base@17.0.8/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base@17.0.8/java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:411)
	at java.base@17.0.8/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
	at java.base@17.0.8/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base@17.0.8/java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:411)
	at java.base@17.0.8/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
	at java.base@17.0.8/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base@17.0.8/java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:411)
	at java.base@17.0.8/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
	at java.base@17.0.8/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base@17.0.8/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base@17.0.8/java.util.stream.IntPipeline$1$1.accept(IntPipeline.java:180)
	at java.base@17.0.8/java.util.stream.Streams$RangeIntSpliterator.forEachRemaining(Streams.java:104)
	at java.base@17.0.8/java.util.Spliterator$OfInt.forEachRemaining(Spliterator.java:711)
	at java.base@17.0.8/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base@17.0.8/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base@17.0.8/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base@17.0.8/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base@17.0.8/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base@17.0.8/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base@17.0.8/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base@17.0.8/java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:411)
	at java.base@17.0.8/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base@17.0.8/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base@17.0.8/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base@17.0.8/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base@17.0.8/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base@17.0.8/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.levelgen.placement.PlacedFeature.m_226368_(PlacedFeature.java:55)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.levelgen.placement.PlacedFeature.redirect$cpm000$onPlaceWithBiome(PlacedFeature.java:547)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.levelgen.placement.PlacedFeature.m_226377_(PlacedFeature.java:44)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.chunk.ChunkGenerator.m_213609_(ChunkGenerator.java:466)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.chunk.ChunkStatus.m_223313_(ChunkStatus.java:145)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.chunk.ChunkStatus$$Lambda$3795/0x00007f7e74c76d60.m_214024_(Unknown Source)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.chunk.ChunkStatus.m_223279_(ChunkStatus.java:292)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ChunkMap.lambda$scheduleChunkGeneration$25(ChunkMap.java:621)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ChunkMap$$Lambda$36938/0x00007f7e77f64780.apply(Unknown Source)
	at MC-BOOTSTRAP/datafixerupper@5.0.28/com.mojang.datafixers.util.Either$Left.map(Either.java:38)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ChunkMap.lambda$scheduleChunkGeneration$27(ChunkMap.java:619)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ChunkMap$$Lambda$36933/0x00007f7e77f61120.apply(Unknown Source)
	at java.base@17.0.8/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150)
	at java.base@17.0.8/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ChunkTaskPriorityQueueSorter.m_143188_(ChunkTaskPriorityQueueSorter.java:62)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ChunkTaskPriorityQueueSorter$$Lambda$34424/0x00007f7e77cae5d8.run(Unknown Source)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.util.thread.ProcessorMailbox.m_18759_(ProcessorMailbox.java:91)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.util.thread.ProcessorMailbox.m_18747_(ProcessorMailbox.java:146)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.util.thread.ProcessorMailbox.run(ProcessorMailbox.java:102)
	at java.base@17.0.8/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
	at java.base@17.0.8/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base@17.0.8/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base@17.0.8/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base@17.0.8/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base@17.0.8/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
```

**Comment by someaddons** [comment]
Fixed in 6.8, should be on cf soon
